runtime/src/types.rs

```diff
@@ -149,10 +162,75 @@ mod currency_id {
             }
         }
     }
-}
 
-pub trait PrettyPrint {
-    fn pretty_print(&self) -> String;
+    #[derive(Debug, Clone, PartialEq)]
+    pub struct RuntimeCurrencyId {
+        pub(crate) currency_id: CurrencyId,
+        pub(crate) asset_metadata: Option<AssetMetadata>,
```

Comment:
I don't really like this.. At first I thought you made this optional because it's only required for ForeignAssets. But then later I saw that you also set this to None when you receive a CurrencyId in e.g. an event, making all the functions that act on it fallible, depending on how and where they were initialized. I'd rather use the real `CurrencyId` in cases where the `RuntimeCurrencyId`  functions will fail for foreign assets